website_sale_hide_price/static/src/xml/website_sale_templates.xml

```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<templates>
+
+<!-- We can't rely on the general setting when specific product have hidden prices-->
+<t t-extend="website_sale.productsSearchBar.autocomplete">
+    <t t-jquery="div[t-if='widget.displayPrice']" t-operation="attributes">
+        <attribute
+                name="t-if"
+                value="widget.displayPrice &amp; !product.website_hide_price"
+            />
+    </t>
+</t>
+<t t-extend="website_sale.productsRecentlyViewed">
+    <t t-jquery="div[t-raw='product.price']" t-operation="attributes">
+        <attribute name="t-if" value="!product.website_hide_price" />
```

Comment:
You are just considering when the product has the check.

In next image you can see it:
![imagen](https://user-images.githubusercontent.com/35952655/170484749-c7671050-5480-4dfd-a4c6-d277e2ec13f6.png)

The partner of Mitchell Admin has the prices hidden and the product Customizable Desk too. When I go to see the recently products, I can see the prices of other products. (You can check it on runboat)

Reply:
Ok, I found a way to fix it :) 

You can review now